[Fix] blog tag warnings and truncation to address build warnings

0xBora opened the pull request:
## Checklist

- [x] I have read the [How to Contribute](https://developers.cardano.org/docs/portal-contribute/).
- [x] I have run `yarn build` after adding my changes **without getting any errors**. 
- [x] I have not committed any changes to `yarn.lock` (or have [removed these changes](https://developers.cardano.org/docs/portal-contribute/#i-committed-yarnlock-to-my-pr-branch-what-do-i-do)).

## Updating documentation or Bugfix

Fixes build warnings regarding to blogs

## Changes Made

- Fix Blog Tags

11 tag warnings affecting SEO and categorization
Fix case sensitivity (AI → ai, DeFi → defi) and add missing media tag
Files: blog/tags.yml + affected blog posts

 - Add Blog Truncation Markers

8 posts without preview truncation
Add <!-- truncate --> markers to recent posts
Files: Blog posts from 2025
 
 > RATIONALE: Related to #1587

rphair reviewed (approved):
nice... I checked in local build ✅ and it has't been this clean in a while!  Only warnings still remaining are:
- `[WARNING] Docusaurus found broken links!` ...
- `[WARNING] Docusaurus found broken anchors!` ...

0xBora commented:
@rphair will be addresssing them in the following PR too in 5-10 minutes :)

katomm commented on blog/2025-01-17-january.md:
```diff
@@ -7,6 +7,8 @@ tags: [media, Developer Office Hours]
 
 This 12th installment of the Cardano Developer Office Hours is presented by Satya Ranjan, Senior Enterprise Architect at the Cardano Foundation. In this session, Satya discusses how the yaci-devkit simplifies building and testing on a local Cardano dev network, helping developers iterate faster and avoid public testnet limitations. The session highlights key features such as customizable block ti ...
 
+<!-- truncate -->
```
The posts tagged with "media" are very short and not ment to be truncated otherwise you can't see the video when scrolling through https://developers.cardano.org/blog/ which gives an odd experience.

katomm commented on blog/2025-06-23-june.md:
```diff
@@ -2,7 +2,7 @@
 title: "Masumi Interview"
 authors: [denicio]
-tags: [ai]
+tags: [AI]
```
While we did not state this anywhere, tags are typically all lower case, because people are otherwise not consistent in their capitalization and then the build breaks. (same for tags in builder tools and showcase)

katomm commented:
We need some documentation on https://developers.cardano.org/docs/portal-contribute for this

rphair reviewed (commented):
I checked in local build ✅ and everything looks & works great: blog is at least as good as before, with clear explanations of tags & truncation at `/docs/portal-contribute#blog-contributions`.

One improvement for contributors (and probably to keep our CI logs clear as well) would be to add `onUntruncatedBlogPosts: 'ignore'` to the blog plugin; seen in the build process:
```
[WARNING] Docusaurus found blog posts without truncation markers:
...
We recommend using truncation markers (`<!-- truncate -->` or `{/* truncate */}`) in blog posts to create shorter previews on blog paginated lists.
Tip: turn this security off with the `onUntruncatedBlogPosts: 'ignore'` blog plugin option.
```

0xBora commented:
Added `onUntruncatedBlogPosts: 'ignore'` as @rphair mentioned to blog plugin config and fixed tag casing mismatch (`Open source` → `open source`. 

Build warnings should now be cleaner.

rphair reviewed (approved):
Definitely cleaner 🤓 ... only 2 warnings I see in local build now are (outside the scope of this PR):

---

This one has been around for a while:
```
[WARNING] The image at "/home/rphair/archive/build/developer-portal/static/img/nfts/overview_nfts.svg" can't be read correctly. Please ensure it's a valid image.
unsupported file type: undefined (file: /home/rphair/archive/build/developer-portal/static/img/nfts/overview_nfts.svg)
[WARNING] The image at "/home/rphair/archive/build/developer-portal/static/img/nfts/overview_nfts.svg" can't be read correctly. Please ensure it's a valid image.
unsupported file type: undefined (file: /home/rphair/archive/build/developer-portal/static/img/nfts/overview_nfts.svg)
```
Needs periodic tune-up, as last we did here (to my knowledge):

* https://github.com/cardano-foundation/developer-portal/pull/1289
```
Browserslist: browsers data (caniuse-lite) is 8 months old. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
```

---

**p.s.** Fixed by https://github.com/cardano-foundation/developer-portal/pull/1610.
